test/e2e/integration_test.go

```diff
@@ -0,0 +1,72 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+
+	infrav1 "github.com/metal3-io/cluster-api-provider-metal3/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("When testing integration [integration]", func() {
+	It("CI Test Provision", func() {
+        numberOfWorkers       = int(*e2eConfig.GetInt32PtrVariable("WORKER_MACHINE_COUNT"))
+        numberOfControlplane = int(*e2eConfig.GetInt32PtrVariable("CONTROL_PLANE_MACHINE_COUNT"))
+        ciTestProvision()
+        ciTestDeprovision()
+	})
+})
+
+func ciTestProvision() {
+    k8sVersion := e2eConfig.GetVariable("KUBERNETES_VERSION")
+	By("Provision Workload cluster")
+	targetCluster = createTargetCluster(k8sVersion)
+	By("Pivot objects to target cluster")
+	pivoting(ctx, func() PivotingInput {
+		return PivotingInput{
+			E2EConfig:             e2eConfig,
+			BootstrapClusterProxy: bootstrapClusterProxy,
+			TargetCluster:         targetCluster,
+			SpecName:              specName,
+			ClusterName:           clusterName,
+			Namespace:             namespace,
+			ArtifactFolder:        artifactFolder,
+			ClusterctlConfigPath:  clusterctlConfigPath,
+		}
+	})
+	By("Repivot everything back to the source cluster")
+	rePivoting(ctx, func() RePivotingInput {
+		return RePivotingInput{
+			E2EConfig:             e2eConfig,
+			BootstrapClusterProxy: bootstrapClusterProxy,
+			TargetCluster:         targetCluster,
+			SpecName:              specName,
+			ClusterName:           clusterName,
+			Namespace:             namespace,
+			ArtifactFolder:        artifactFolder,
+			ClusterctlConfigPath:  clusterctlConfigPath,
+		}
+	})
+}
+
+func ciTestDeprovision() {
+	bootstrapClient := bootstrapClusterProxy.GetClient()
+	deprovisionTargetCluster(&bootstrapClient)
+	numberOfAvailableBMHs := 2 * numberOfWorkers
+	intervals := e2eConfig.GetIntervals(specName, "wait-bmh-deprovisioning-available")
+	WaitForNumBmhInState(ctx, bmov1alpha1.StateAvailable, WaitForNumInput{
+		Client:    bootstrapClient,
+		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Replicas:  numberOfAvailableBMHs,
+		Intervals: intervals,
+	})
+	intervals = e2eConfig.GetIntervals(specName, "wait-metal3-machine-deprovisioning")
+	waitForResourceGone(&bootstrapClient, &infrav1.Metal3MachineList{}, namespace, intervals)
+	intervals = e2eConfig.GetIntervals(specName, "wait-machine-deprovisioning")
+	waitForResourceGone(&bootstrapClient, &clusterv1.MachineList{}, namespace, intervals)
+	intervals = e2eConfig.GetIntervals(specName, "wait-metal3-cluster-deprovisioning")
+	waitForResourceGone(&bootstrapClient, &infrav1.Metal3ClusterList{}, namespace, intervals)
+	intervals = e2eConfig.GetIntervals(specName, "wait-cluster-deprovisioning")
+	waitForResourceGone(&bootstrapClient, &clusterv1.ClusterList{}, namespace, intervals)
+}
```

Comment:
I think we can safely use DeleteCluster from the framework instead of this. It will check that all resources are gone. The only thing left is to check that the BMHs become available.